bin/varnishd/cache/cache_vrt.c

```diff
@@ -62,8 +62,22 @@ VRT_synth(VRT_CTX, VCL_INT code, VCL_STRING reason)
 
 	CHECK_OBJ_NOTNULL(ctx, VRT_CTX_MAGIC);
 	assert(ctx->req != NULL || ctx->bo != NULL);
-	if (code < 100 || code > 65535)
-		code = 503;
+	if (code < 0) {
+		VRT_fail(ctx, "return(synth()) status code (%jd) is negative",
+		    code);
+		return;
+	}
+	if (code > 65535) {
+		VRT_fail(ctx, "return(synth()) status code (%jd) > 65535",
+		    code);
+		return;
+	}
+	if ((code % 1000) < 100) {
```

Comment:
I would have thought that setting `beresp.status` and `return(synth(...))` are different things, in that the extended status codes are only valid for the latter.
Sure, at some point we need to error for `status < 100`, but would would think that entering `vcl_synth` with status 1000..10099 should be valid.

Reply:
Since we do VRT_fail immediately when attempting to set a status code to something that is illegal using `set resp.status`, I worry that there would be other code that assumes that a status can not be less than 100. So the safe way seems to me to mimick closely what happens when executing `set resp.status`.

Reply:
> beresp.status and return(synth(...)) are different things, in that the extended status codes are only valid for the latter.

I don't understand. The extended status codes are valid for both of them, so as I understand it these become two ways of achieving the same, which is why I think they should also have the same error conditions.

Reply:
I understand your reasons, but from a user perspective, a `return(synth(1000))` ends up in `resp.status == 1000` in `vcl_synth` and as such would look legit.
Anyway, I will not repeat my argument another time, I think both options have their pros and cons and I think we just need to make a decision here, ideally during bugwash....

Reply:
And maybe _valid_ was a bad word. What I mean is that the extended status codes only _matter_ for `vcl_synth` and can be changed there into valid ones (unless the modulus is what you want).

Reply:
as discussed during bugwash, there is a good reason to make `synth()` consistent with setting `(be)?resp.status`